src/bargain.c

```diff
@@ -0,0 +1,330 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "bargain.h"
+#include "stores.h"
+#include "items_types.h"
+#include "menu.h"
+#include "test_bargain.h"
+#include "api/fetch.h"
+#include "test_functions.h"
+
+#define MAX_STORES_COUNT 3
+#define MAIN_MENU_ITEMS_COUNT 4
+
+typedef int (*compfn)(const void *, const void *);
+
+void bargain_run_bargain()
+{
+
+    char *options[MAIN_MENU_ITEMS_COUNT];
+
+    options[0] = "Find Bargains";
+    options[1] = "Edit Shopping List";
+    options[2] = "Settings";
+    options[3] = "Quit";
+
+    int choice = display_menu(options, "Main Menu", "Enter the number of the option you want to choose, or 4 to quit");
+
+    switch (choice)
+    {
+    case 0:
+        bargain_menu_find_bargain();
+        break;
+    case 1:
+        // TODO: Implement Basket Editor
+        break;
+    case 2:
+        // TODO: Implement Settings
+        break;
+    case 3:
```

Comment:
we don't need to have a dedicated 'quit' option, since that is already handled by returning -1 when the user enters '!q'